agents/official/bee-agent-framework/src/content-judge.ts

```diff
@@ -0,0 +1,281 @@
+    [key in Criteria]: z.ZodNumber;
+  }
+);
+
+// Define weighting for each evaluation criterion (using weighted average),
+// make sure the sum of the weights equals 1
+type Weights = { [key in Criteria]: number };
+const weights: Weights = {
+  correctness: 0.5,
+  depth_and_coverage: 0.1,
+  clarity_and_structure: 0.1,
+  relevance: 0.3,
+};
+
+const EVALUATION_PROMPT = `Evaluate the quality of the generated document based on multiple criteria using a continuous scale from 0 (lowest quality) to 1 (highest quality). For each criterion, provide a numerical score (0-1):
+- correctness: Assess whether the information is factually accurate based on reliable, authoritative sources. Penalize factual errors, misinterpretations, or unverified claims. Prioritize primary sources and well-established research when cross-checking.
+- depth_and_coverage: Evaluate whether the document provides a comprehensive and well-rounded discussion of the topic. Consider whether key aspects are addressed, nuances are explored, and supporting evidence is provided. Compare the breadth and depth of the information against expectations for the topic.
+- clarity_and_structure: Assess how well the document is organized. Check for logical flow, appropriate use of headings, bullet points, summaries, and coherence. Consider whether the language is clear, precise, and accessible to the intended audience.
+- relevance: Measure the alignment between the document and the given research prompt or problem. Determine if the content remains focused on relevant aspects, avoiding tangents, unnecessary details, or off-topic discussions.`;
+
+const calculateScore = (result: Weights) =>
+  // Multiply by 100 and round to avoid floating precision problem when comparing
+  Math.round(
+    criteria.reduce((sum, key) => sum + result[key] * weights[key] * 100, 0)
+  );
+
+const retrieveDocuments = async ({
+  prompt,
+  agents,
+}: {
+  prompt: string;
+  agents: string[];
+}) => {
+  const client = new ACPClient(
+    {
+      name: "example-client",
+      version: "1.0.0",
+    },
+    {
+      capabilities: {
+        prompts: {},
+        resources: {},
+        tools: {},
+      },
+    }
+  );
+  const transport = new SSEClientTransport(
+    new URL("/mcp/sse", "http://localhost:8333")
+  );
+
+  try {
+    await client.connect(transport);
+  } catch (error) {
+    console.error("Error connecting to ACP server:", error);
+    throw error;
+  }
+
+  const { agents: platformAgents } = await client.listAgents();
+  const platformAgentsName = platformAgents.map((agent) => agent.name);
+
+  if (!agents.every((agent) => platformAgentsName.includes(agent))) {
+    throw new Error("One or more agents not available in the platform");
+  }
+
+  const results = await Promise.all(
+    agents.map((agent) =>
+      client.runAgent(
+        {
+          name: agent,
+          input: { prompt },
+        },
+        {
+          timeout: 10 * 60 * 1000,
+        }
```

Comment:
Prefer passing signal, that gives control to the user rather than fixed timeout.